### ISSUE TITLE
KeyboardAvoidingView bug on TransitionStart

### ISSUE THREAD

**Issue by danitatt** [software-mansion/react-native-screens#958]
## Description
Apparently due to the loss of focus on the screen, the KeyboardAvoidingView changes its state. See screenshots for more details. If I use event TransitionStart and call Keyboard.dismiss, then nothing happens while I swipe the screen. Is it possible to somehow solve this problem?

### Screenshots
![GitHub Logo](https://i.ibb.co/CH9Xsgc/1.gif)
![GitHub Logo](https://i.ibb.co/MG4RVT0/2.gif)

## Package versions
```javascript
{
  "@react-navigation/native": "^5.9.4",
  "react-native-screens": "^3.3.0",
  "react": "17.0.1",
  "react-native": "0.64.1"
}
```

**Comment by kacperkapusciak** [comment]
Hi @danitatt 👋 
Could you provide a minimal reproducible code example? Could be as an Expo Snack or just a code-snippet. It would be very helpful.

**Comment by danitatt** [comment]
@kacperkapusciak I'm sorry I didn't do this before 😢
Its a minimal clean code for example [Snack](https://snack.expo.io/@danitatt/keyboardavoidingview-bug-on-transitionstart)

**Comment by WoLewicki** [comment]
@danitatt can you check if #970 fixes your problem? I am not sure if it is a correct solution, but I couldn't find out why calling `Keyboard.dismiss()` is not enough, even though it correctly triggers native code, same as `stack` navigator.

**Comment by danitatt** [comment]
@WoLewicki no, this not fix problem =/. The keyboard dismiss after when the screen comes back. With this fix, it began to seem more bugs 😔😔
